listen.go

```diff
@@ -26,20 +28,20 @@ func (r *Relay) Matches(a ma.Multiaddr) bool {
 }
 
 func (l *RelayListener) Accept() (tpt.Conn, error) {
-	ctx := l.Relay().ctx
 	select {
 	case c := <-l.incoming:
-		log.Infof("accepted relay connection: %s", c.ID())
-		s := RelayStatus{
-			Code:    StatusOK,
-			Message: "OK",
-		}
-		if err := s.WriteTo(c); err != nil {
+		err := l.Relay().writeResponse(c.Stream, pb.CircuitRelay_SUCCESS)
+		if err != nil {
+			log.Debugf("error writing relay response: %s", err.Error())
+			c.Stream.Close()
```

Comment:
But we do want the other side to be able to continue to write, don't we?
This is a transparent proxy in effect, the other side's writes will continue to be forwarded by the second goroutine until it also closes the channel.